### PR TITLE
doc: added section for `customUserProperties.update`

### PR DESCRIPTION
**Pull request by ledo01**

### CODE REVIEW

**Comment by ledo01** [comment]
Copy is not one of my talent, so lmk if you have a better one!

**Comment by ledo01** [comment]
[[0.5d] new SDK command: "add custom user prop"](https://trello.com/c/3fq6i01G/12742-05d-new-sdk-command-add-custom-user-prop)

File: source/index.html.md [hunk 1]
```diff
@@ -309,3 +309,28 @@ Always pass in the most up-to-date value for <code>total_paid</code> as a user's
 <aside class='notice'>
   Updates to orders will not be saved until the user who created the order is "authenticated" via the SDK. In practice, this means that a <code>ticketingOrder.update</code> command will be processed immediately for users who have been "autenticated" by a previous <code>emailSignup</code> or <code>phoneNumberSignup</code> command. For users who weren't "authenticated" before a <code>ticketingOrder.update</code> command is made, the <code>ticketingOrder.update</code> command will be "buffered" by the SDK and processed immediately following a <code>emailSignup</code> or <code>phoneNumberSignup</code> command (which can potentially happen in the future on a separate pageview or session entirely).
 </aside>
+
+# Add Custom User Properties
```
Comment (prateekmadhikar) [inline source/index.html.md]:
```suggestion
# Set Custom User Properties
```

File: source/index.html.md [hunk 2]
```diff
@@ -309,3 +309,28 @@ Always pass in the most up-to-date value for <code>total_paid</code> as a user's
...
+
+This command will add and update custom user properties for the current user.
```
Comment (prateekmadhikar) [inline source/index.html.md]:
```suggestion
This command will set custom user properties on the currently "authenticated" user. Any custom user properties that did not exist before will get created and any existing custom user properties that do exist will be updated.
```

File: source/index.html.md [hunk 3]
```diff
@@ -309,3 +309,28 @@ Always pass in the most up-to-date value for <code>total_paid</code> as a user's
 <aside class='notice'>
   Updates to orders will not be saved until the user who created the order is "authenticated" via the SDK. In practice, this means that a <code>ticketingOrder.update</code> command will be processed immediately for users who have been "autenticated" by a previous <code>emailSignup</code> or <code>phoneNumberSignup</code> command. For users who weren't "authenticated" before a <code>ticketingOrder.update</code> command is made, the <code>ticketingOrder.update</code> command will be "buffered" by the SDK and processed immediately following a <code>emailSignup</code> or <code>phoneNumberSignup</code> command (which can potentially happen in the future on a separate pageview or session entirely).
 </aside>
+
+# Add Custom User Properties
+
+This command will add and update custom user properties for the current user. 
+
+```javascript
```
Comment (prateekmadhikar) [inline source/index.html.md:319]: i believe we need to add this to add a header for the code block!

image

```suggestion
> Use the following command to set custom user properties on the authenticated user.

```javascript
```

File: source/index.html.md (hunk 2 above)
Comment (prateekmadhikar) [reply source/index.html.md]: lets also move this below the code block - i think thats necessary to do!

File: source/index.html.md [hunk 4]
```diff
@@ -309,3 +309,28 @@ Always pass in the most up-to-date value for <code>total_paid</code> as a user's
...
+    // success calback, called after data is saved
+  } 
+  function(data) {
+    // failure callback, called if something goes wrong
+    // error information is provided in the "data" param
+  }
+)
+```
+
+The third argument is an object of the custom properties name as keys with their values as the object values. This call will created any non-existing properties and update the value of any existing one.
```
Comment (prateekmadhikar) [inline source/index.html.md]: I think someone who reads this documentation can understand it fairly quickly and easily without this blurb! Maybe, instead, we can let the reader know that the value's must be strings and if not they will get converted to strings when saved onto the user (its a db design thing.. lowkey sucks lol)

```suggestion
Note that the values for each custom user property must be a string. Any value passed in that is not a string will be saved as a string.
```

**Review by prateekmadhikar — CHANGES_REQUESTED** [review]
Few things to address! And:

- Lets add a blurb like we have in other sections to let users know that this wont work until the user is authenticated. We can use this copy:

```
<aside class='notice'>
  Users will not be have their custom user properties set or updated until they're "authenticated" via the SDK. In practice, this means that a <code>customUserProperties.update</code> command will be processed immediately for users who have been "autenticated" by a previous <code>emailSignup</code> or <code>phoneNumberSignup</code> command. For users who weren't "authenticated" before a <code>customUserProperties.update</code> command is made, the <code>customUserProperties.update</code> command will be "buffered" by the SDK and processed immediately following a future <code>emailSignup</code> or <code>phoneNumberSignup</code> command (which can potentially happen in the future on a separate pageview or session entirely).
</aside>
```

- Can you double check that the formatting looks fine? I ran the `docker compose` and this branch (and master too) looks weird locally but it looks fine on prod? Pls confirm all is fine on prod (sdk.hive.co) after you ship and deploy!

image

**Comment by ledo01** [comment]
> Can you double check that the formatting looks fine? I ran the docker compose and this branch (and master too) looks weird locally but it looks fine on prod? Pls confirm all is fine on prod (sdk.hive.co) after you ship and deploy!

I think it's because we are running an older version of slate than the one used in the container. Looks like we are on v2.0.0 and, unfortunately, this version isn't available on docker hub and not worth the trouble of making one. 

I'll make sure to check on prod thought!

**Review by prateekmadhikar — APPROVED** [review]
lgtome!

**Pull request merged**